### DISCUSSION TITLE
Issues Running Headless Chrome Selenium Driver in Vercel Serverless Function

### DISCUSSION THREAD

**Discussion by cnemri** [vercel/community#4851] [Bug Report]
### Summary

Hello Vercel Community,

I'm encountering a persistent issue when trying to run a headless Chrome Selenium driver within a Flask Python serverless function on Vercel. My aim is to scrape data from a URL using Selenium, which works perfectly in my local environment. However, when deploying on Vercel, I consistently face the following error:

`selenium.common.exceptions.WebDriverException: Message: Service chromedriver/chromedriver unexpectedly exited. Status code was: 127
`

**Attempts to resolve**

- Ensured that the ChromeDriver is accessible within the function.
- Verified and set execute permissions for the ChromeDriver.
- Using the Linux64 ChromeDriver (URL: [ChromeDriver - Linux64](https://edgedl.me.gvt1.com/edgedl/chrome/chrome-for-testing/119.0.6045.105/mac-arm64/chromedriver-mac-arm64.zip)) compatible with the operating system.
- Created a ChromeService specifying the path to the ChromeDriver before building the actual driver.
- Configured the ChromeService to log to /tmp to avoid read-only filesystem restrictions.

**Relevant code snippet**

    os.system("cp chromedriver/chromedriver /tmp/chromedriver")
    options = Options()
    options.add_argument("--headless=new")
    service = webdriver.ChromeService(executable_path='/tmp/chromedriver',log_output='/tmp/chromedriver.log')
    driver = webdriver.Chrome(service=service,options=options)

Environment Details:

- Flask app with an API route responsible for scraping data using Selenium.
- Everything functions as expected locally.
- I would greatly appreciate any insights or suggestions on how to resolve this issue. Has anyone faced a similar problem or can guide me on running headless Chrome Selenium in a serverless function on Vercel?

Thank you in advance for your assistance!

### Steps to Reproduce

    os.system("cp chromedriver/chromedriver /tmp/chromedriver")
    options = Options()
    options.add_argument("--headless=new")
    service = webdriver.ChromeService(executable_path='/tmp/chromedriver',log_output='/tmp/chromedriver.log')
    driver = webdriver.Chrome(service=service,options=options)

**Comment by davisonv** [comment]
I'll do a similar project this weekend, so i'm eager to know if vercel's serverless function has suport for selenium

**Comment by onacitarhan17** [comment]
Hey, any updates on this?

**Comment by sshah98** [comment]
any update ont this?

**Comment by closebridge** [comment]
Any updates on this so far?
